app/src/main/java/org/fossasia/phimpme/gallery/activities/LFMainActivity.java

```diff
@@ -651,27 +653,42 @@ private void setDrawerTheme() {
         findViewById(R.id.ll_drawer_Setting).setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                Intent intent = new Intent(LFMainActivity.this, SettingsActivity.class);
                 mDrawerLayout.closeDrawer(GravityCompat.START);
-                startActivity(intent);
+                new Handler().postDelayed(new Runnable() {
+                    @Override
+                    public void run() {
+                        Intent intent = new Intent(LFMainActivity.this, SettingsActivity.class);
+                        startActivity(intent);
+                    }
+                }, Constants.NAVIGATION_DRAWER_EXIT_TIME);
```

Comment:
@sridharjajoo Instead of using a handler with hard-coded drawer exit time, why don't you use a drawer listener?

Reply:
I initially used onDrawerClosed listener for implementing the solutoion but it is only available for API
level 22 and higher.So in cases where API level is less than 22 a lag will still be there.

Reply:
Should I send a PR which uses drawer listener?

Reply:
@sridharjajoo yes, please.

Reply:
@vinaysajjanapu 
I am sending a new PR . Kindly close this.
Thank You